### DISCUSSION TITLE
unique interface names in devices?

### DISCUSSION THREAD

**Discussion by steffenschumacher** [netbox-community/netbox#14462] [Help Wanted!] [answered]
I've enabled ENFORCE_GLOBAL_UNIQUE, but that doesn't prevent duplicate interfaces from being inserted - in which cases is it desirable to have duplicate interfaces for a device?
Is there a way to prevent duplicate interfaces from being inserted/created?

**Comment by jeremystretch** [comment, marked as answer]
The [`ENFORCE_GLOBAL_UNIQUE`](https://docs.netbox.dev/en/stable/configuration/miscellaneous/#enforce_global_unique) parameter applies to IP addressing; it has nothing to do with devices or interfaces.

NetBox requires that all interfaces (and other components) assigned to a device have a unique name.

**Reply by steffenschumacher** [reply]
Mystery solved..
This particular case is on a device in a VirtualChassis - and if I view the primary device in the stack, then I'm actually seeing all the interfaces of the virtual chassis and not just the ones from the primary device.
Once used the dcim.interfaces API, filtering on device_id I have no duplicates..
I can work with this now that I know what's happening - thanks to @mtinberg for the hint!

**Comment by candlerb** [comment]
> doesn't prevent duplicate interfaces from being inserted

Can you give a specific, step-by-step way to reproduce what you're seeing? And please say what version of Netbox you're using.

AFAIK, all versions of Netbox in living memory have enforced a requirement that the interface name on a device is unique across that device.  Of course, different devices can have interfaces with the same name, like "eth0"

**Comment by mtinberg** [comment]
That may be technically correct but from a user perspective it's not always true, stacked Virtual Chassis devices can have different members with the same interface names such that when the primary device is viewed show dup interface names, eg. when you have a workflow dependent on renaming interfaces on stacked devices after creating them that goes awry.  In the DB the uniqueness constraint may be satisfied, but in the UI it might not be.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
